pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -319,14 +319,19 @@ stdenv.mkDerivation {
                       && targetPlatform.isLinux
                       && !(stdenv.targetPlatform.useAndroidPrebuilt or false)
                       && !(stdenv.targetPlatform.useLLVM or false)
-                      && gccForLibs != null) ''
+                      && gccForLibs != null) (''
       echo "--gcc-toolchain=${gccForLibs}" >> $out/nix-support/cc-cflags
 
       # Pull in 'cc.out' target to get 'libstdc++fs.a'. It should be in
       # 'cc.lib'. But it's a gcc package bug.
       # TODO(trofi): remove once gcc is fixed to move libraries to .lib output.
       echo "-L${gccForLibs}/${optionalString (targetPlatform != hostPlatform) "/${targetPlatform.config}"}/lib" >> $out/nix-support/cc-ldflags
     ''
+    # this ensures that when clang passes -lgcc_s to lld (as it does
+    # when building e.g. firefox), lld is able to find libgcc_s.so
```

Comment:
If you drop the backreference symlinks below:

```
      ln -s $libgcc/lib/libgcc_s.so   $lib/lib/
      ln -s $libgcc/lib/libgcc_s.so.1 $lib/lib/
```

then `-L${gccForLibs.libgcc}/lib"` would have a natural presence for most packages. Is it a fair statement?

Reply:
Hrm, dropping the symlinks and using `-L${gccForLibs.libgcc}/lib` doesn't work, but the reasons are not yet clear.  I will investigate this tomorrow.

Reply:
Follow-up on this?

Reply:
See https://github.com/NixOS/nixpkgs/pull/209870#issuecomment-1493235853

Resolving for now (deferred as https://github.com/NixOS/nixpkgs/issues/221525)